scRNA-seq/setup/mouse-liver/prepare_mouse_liver_markers.Rmd

```diff
@@ -0,0 +1,326 @@
+```
+
+Plot PC1 and PC2
+
+```{r}
+scater::plotReducedDim(normalized_sce, "PCA")
+```
+
+### Clusters
+
+We'll perform clustering on the PCA matrix.
+
+```{r}
+# Extract the principal components matrix
+pca_mat <- reducedDim(normalized_sce, "PCA")
+```
+
+Cluster the cells in PCA space, using graph-based clustering.
+
+```{r}
+# Run the clustering algorithm
+nnclusters <- bluster::clusterRows(
+  pca_mat, 
+  bluster::NNGraphParam(k = 10, 
+                        cluster.fun = "louvain")
+)
+# Store cluster results in the SCE object
+normalized_sce$nncluster <- factor(nnclusters)
+```
+
+#### Visualize clusters
+
+We'll use UMAP for visualization here.
+
+```{r}
+# Run UMAP
+normalized_sce <- scater::runUMAP(normalized_sce, 
+                                  dimred = "PCA")
+```
+
+And now for viz...
+
+```{r}
+scater::plotReducedDim(normalized_sce, 
+                       "UMAP", 
+                       colour_by = "nncluster", 
+                       text_by = "nncluster")
+```
+
+Save this plot to use in the exercise notebook itself.
+
+```{r}
+ggplot2::ggsave(umap_file, plot = ggplot2::last_plot())
+```
+
+Write the normalized and transformed object to file, complete with PCA, UMAP, and cluster labels.
+We will provide this RDS file during the training workshop.
+
+```{r}
+readr::write_rds(normalized_sce, file = normalized_file)
+```
+
+## Marker genes
+
+Our main goal for this notebook is to obtain tables of marker genes for the exercise.
+We'll use `pval.type = "all"` to mirror what we cover during instruction.
+
+```{r}
+markers <- scran::findMarkers(normalized_sce, 
+                              groups = normalized_sce$nncluster, 
+                              pval.type = "all")
+```
+
+Let's take a look at the top gene rankings for cluster 7.
+
+```{r}
+head(markers[[7]], n = 10)
+```
+
+Jchain has a small FDR value and negative summary log fold change; let's take a look at the expression values in the UMAP plot.
+
+```{r}
+# Plot UMAP results colored by expression
+scater::plotReducedDim(normalized_sce, "UMAP", 
+                       colour_by = "Jchain")
+```
+Visually, it seems like Jchain has an above average expression value in cluster 7.
+Summary log fold change is a collapsing the effect sizes into a single (summarized!) value.
+From the table and the plot, we can see that the negative summary log fold change is probably driven by the very high expression values for Jchain in cluster 4.
+
+Saa3 has a positive summary log fold change, let's take a look.
+
+```{r}
+# Plot UMAP results colored by expression
+scater::plotReducedDim(normalized_sce, "UMAP", 
+                       colour_by = "Saa3")
+```
+Though we see some individual samples with high expression values for Saa3 in cluster 2, it looks like it is most consistently highly expressed in cluster 7.
+
+Now we're ready to write each table of marker genes, sorted by FDR, to file.
+
+```{r}
+# Assign, because otherwise this is _chatty_
+marker_df_list <- purrr::imap(
```

Comment:
Thing I learned recently: `purrr::iwalk()` (& `purrr::walk()`, and basically any map->walk replacement) do the side effects without returning anything. So since you are not using the assignment, you can do this:
```suggestion
purrr::iwalk(
```